File: ElitTournamentWeb.Angular/app/src/app/core/models/post/update-many-post-request.ts
import { PostType } from '../types/post-type';

export class UpdateManyPostRequest {
    public posts: Array<UpdateManyPostItem>;
    public removedPosts: Array<UpdateManyPostItem>;

    constructor(posts: Array<UpdateManyPostItem>, removedPosts: Array<UpdateManyPostItem>) {
        this.posts = posts;
        this.removedPosts = removedPosts;
    }
}

export class UpdateManyPostItem {
    public id: number;
    public type: PostType;
    public title: string;
    public description: string;
    public text: string;
    public image: string;
}

File: ElitTournamentWeb.ViewModels/Post/UpdateManyPostRequest.cs
using System.Collections.Generic;

namespace ElitTournamentWeb.ViewModels.Post
{
	public class UpdateManyPostRequest
	{
		public IEnumerable<UpdateManyPostItemRequest> Posts { get; set; }
		public IEnumerable<UpdateManyPostItemRequest> RemovedPosts { get; set; }
	}

	public class UpdateManyPostItemRequest
	{
		public int Id { get; set; }
		public int Type { get; set; }
		public string Title { get; set; }
		public string Description { get; set; }
		public string Text { get; set; }
		public string Image { get; set; }
	}
}